### PR TITLE
adds a relay token filter to gateway

### LINKED ISSUE #3326: Migration to Spring Cloud RC1 makes JWT token fail in microservices
This is in the master branch, it looks like since we upgraded to the latest Spring Cloud RC1 the JWT tokens are not working anymore (for our 3.0.0 release we were using Spring Cloud M5).

This is because the "Authorization" header is viewed as sensitive (in the ZuulProperties), so it is not forwarded anymore to the microservices.

### PR DESCRIPTION
**Pull request by xetys**
this removes the authorization header from zuuls ignore list

fixes #3326

### CODE REVIEW

**Comment by xetys** [comment]
is this caused by me or some bug occured in travis?

**Comment by gmarziou** [comment]
I don't think so, looks like Travis's hiccup.
Let's close and reopen the PR, this should restart the build.

**Comment by xetys** [comment]
my local test with https://github.com/xetys/jhipster-travis-helper was succeeding :smile:

**Comment by gmarziou** [comment]
This is really great tool, I'll use it.

**Comment by jdubois** [comment]
OK that's simpler than what I wanted to do for #3326 with an "internal JWT token", so that's better.

I'll merge this, but I also found some other minor issues while working on #3326 (mostly formatting or style problem), so I'll do everything at once.

**Comment by wmarques** [comment]
I tested it and it works.

File: generators/server/templates/src/main/java/package/gateway/_TokenRelayFilter.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.util.Set;
+
+@Component
+public class TokenRelayFilter extends ZuulFilter {
+    @Override
+    public Object run() {
+        RequestContext ctx = RequestContext.getCurrentContext();
+
+        // Alter ignored headers as per: https://gitter.im/spring-cloud/spring-cloud?at=56fea31f11ea211749c3ed22
```
Comment (jdubois) [inline generators/server/templates/src/main/java/package/gateway/_TokenRelayFilter.java]: I'll change the comment as gitter is not persistent (messages are lost after 15 days)

**Comment by xetys** [comment]
done!

File: generators/server/templates/src/main/java/package/gateway/_TokenRelayFilter.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+@Component
+public class TokenRelayFilter extends ZuulFilter {
+    @Override
+    public Object run() {
+        RequestContext ctx = RequestContext.getCurrentContext();
+
+        Set<String> headers = (Set<String>) ctx.get("ignoredHeaders");
+        // We need our JWT tokens relayed to resource servers
+        headers.remove("authorization");
```
Comment (jdubois) [inline generators/server/templates/src/main/java/package/gateway/_TokenRelayFilter.java]: Shouldn't this be "Authorization"? Is this case-sensitive?

**Comment by xetys** [comment]
it looks like an config value...sinse there is a list of ignored headers.....so this is not EXACTLY the header value

and travis is kidding me again :smile:

**Pull request merged**